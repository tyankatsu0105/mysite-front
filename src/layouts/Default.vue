<template>
	<div class="Default">
		<VHeader />
		<transition-group
			name="page"
			tag="main"
			appear
		>
			<div
				key="main"
				class="Default_Contents"
			>
				<slot></slot>
			</div>
		</transition-group>
		<VFooter />
	</div>
</template>

<script>
import VHeader from "@/containers/VHeader.vue";
import VFooter from "@/containers/VFooter.vue";
export default {
  components: {
    VHeader,
    VFooter
  },
  metaInfo: {
    htmlAttrs: {
      lang: "ja"
    }
  }
};
</script>

<style lang="scss" scoped>
@import "@/styles/utility/transition-page.scss";
.Default {
  display: grid;
  grid-template-rows: 1fr 50px;
  grid-template-columns: auto;
  min-height: 100vh;

  &_Contents {
    max-width: 1500px;
    padding: 56px 36px;
    margin: 0 auto;

    @include mq-xs {
      max-width: 100vw;
      padding: 24px 15px;
      margin: 88px auto 24px;
    }
  }
}
</style>
