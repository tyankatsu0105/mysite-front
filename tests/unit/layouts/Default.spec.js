import { shallowMount } from "@vue/test-utils";
import Default from "@/layouts/Default.vue";

describe("Default", () => {
  it("render", () => {
    const wrapper = shallowMount(Default);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
