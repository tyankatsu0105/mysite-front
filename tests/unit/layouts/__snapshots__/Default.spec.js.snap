// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default render 1`] = `
<div class="Default">
  <vheader-stub></vheader-stub> <span name="page" tag="main" appear=""><div class="Default_Contents"></div></span>
  <vfooter-stub></vfooter-stub>
</div>
`;
